### PR TITLE
feature: preferred token type

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
+ fix var_export warning
+ update changelog and docs

### CODE REVIEW

File: lib/Auth/Process/PrivacyideaAuthProc.php [hunk 1]
```diff
@@ -79,10 +80,10 @@ public function process(&$state)
...
         // If set in config, allow to check the IP of the client and to control the 2FA depending on the client IP.
         // It can be used to configure that a user does not need to provide a second factor when logging in from the local network.
-        if (!empty($this->authProcConfig['excludeClientIPs']))
-        {
-            $state['privacyIDEA']['enabled'][0] = $this->matchIP(sspmod_privacyidea_Auth_utils::getClientIP(), $this->authProcConfig['excludeClientIPs']);
-        }
+//        if (!empty($this->authProcConfig['excludeClientIPs']))
+//        {
+//            $state['privacyIDEA']['enabled'][0] = $this->matchIP(sspmod_privacyidea_Auth_utils::getClientIP(), $this->authProcConfig['excludeClientIPs']);
+//        }
```
Comment (nilsbehlen) [inline lib/Auth/Process/PrivacyideaAuthProc.php:86]: Why is this commented?

File: docs/privacyidea.md [hunk 2]
```diff
@@ -223,6 +230,13 @@ If you want to use privacyIDEA as an auth proc filter, change the metadata. Use
          */
         'SSO' => 'false',
         
+        /**
+         * Set preferredTokenType to your favourite token type.
+         * If the choosed token will be triggered,
```
Comment (nilsbehlen) [inline docs/privacyidea.md:235]: token **is** triggered

File: docs/privacyidea.md [hunk 3]
```diff
@@ -87,11 +87,18 @@ You need to add the authentication source 'privacyidea' to
...
-     * Set this to 'true' if you want to use single sign on.
+     * Set SSO to 'true' if you want to use single sign on.
      * All information required for SSO will be saved in the session.
      * After logging out, the SSO data will be removed from the session.
      */
     'SSO' => 'false',
+    
+    /**
+     * Set preferredTokenType to your favourite token type.
+     * If the choosed token will be triggered,
```
Comment (nilsbehlen) [inline docs/privacyidea.md:98]: token **is** triggered, also i think no new is required

File: lib/Auth/Process/PrivacyideaAuthProc.php (hunk 1 above)
Comment (lukasmatusiewicz) [reply lib/Auth/Process/PrivacyideaAuthProc.php]: fixed in #409b613

File: docs/privacyidea.md (hunk 3 above)
Comment (lukasmatusiewicz) [reply docs/privacyidea.md:98]: I don't understand what exactly you mean. Can you change it by yourself?

**Pull request merged**